samples/server/petstore/nancyfx/src/IO.Swagger/Modules/UserModule.cs

```diff
@@ -0,0 +1,221 @@
+using System;
+using Nancy;
+using Nancy.ModelBinding;
+using System.Collections.Generic;
+using Sharpility.Base;
+using IO.Swagger.v2.Models;
+using IO.Swagger.v2.Utils;
+using NodaTime;
+
+namespace IO.Swagger.v2.Modules
+{ 
+
+    /// <summary>
+    /// Module processing requests of User domain.
+    /// </summary>
+    public sealed class UserModule : NancyModule
+    {
+        /// <summary>
+        /// Sets up HTTP methods mappings.
+        /// </summary>
+        /// <param name="service">Service handling requests</param>
+        public UserModule(UserService service) : base("/v2")
+        { 
+            Post["/user"] = parameters =>
+            {
+                var body = this.Bind<User>();
+                service.CreateUser(Context, body);
+                return new Response { ContentType = "application/json"};
+            };
+
+            Post["/user/createWithArray"] = parameters =>
+            {
+                var body = this.Bind<List<User>>();
+                service.CreateUsersWithArrayInput(Context, body);
+                return new Response { ContentType = "application/json"};
+            };
+
+            Post["/user/createWithList"] = parameters =>
+            {
+                var body = this.Bind<List<User>>();
+                service.CreateUsersWithListInput(Context, body);
+                return new Response { ContentType = "application/json"};
+            };
+
+            Delete["/user/{username}"] = parameters =>
+            {
+                var username = Parameters.ValueOf<string>(parameters, Context.Request, "username", ParameterType.Path);
```

Comment:
@jchannon I'm not the author so I don't know the exact reason.

My guess is that `username` can be of different type (defined in swagger spec), e.g. date, double, string, etc and that's why it needs to call `Parameters.ValueOf<datatype>` to cast the value into the type defined in the spec.

Ref: https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/resources/nancyfx/api.mustache#L29